unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/States/EditorMode/Voxels/VoxelController.cs

```diff
@@ -0,0 +1,305 @@
+using DCL.Configuration;
+using DCL.Controllers;
+using DCL.Models;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class VoxelController : MonoBehaviour
+{
+    [Header("References")]
+    public VoxelPrefab voxelPrefab;
+    public BuilderInputWrapper builderInputWrapper;
+    public BuildModeController buildModeController;
+    public BuildEditorMode buildEditorMode;
+    public OutlinerController outlinerController;
+    public FreeCameraMovement freeCameraMovement;
+    
+    public LayerMask groundLayer;
+
+    DecentralandEntityToEdit lastVoxelCreated;
+
+    GameObject editionGO;
+    bool mousePressed = false, isVoxelModelActivated = false,isCreatingMultipleVoxels = false; 
+    Vector3Int lastVoxelPositionPressed;
+    Vector3 lastMousePosition;
+    Dictionary<Vector3Int, VoxelPrefab> createdVoxels = new Dictionary<Vector3Int, VoxelPrefab>();
+    ParcelScene currentScene;
+
+    const float RAYCAST_MAX_DISTANCE = 10000f;
+
+    private void Start()
+    {
+        builderInputWrapper.OnMouseDown += MouseDown;
+        builderInputWrapper.OnMouseUp += MouseUp;
+ 
+    }
+
+    private void Update()
+    {
+        if (mousePressed && isVoxelModelActivated && isCreatingMultipleVoxels)
+        {
+
+            bool fillVoxels = false;
```

Comment:
I'd say we should remove this flag and just call `FillVoxels(lastVoxelPositionPressed, currentPosition);` at the end of each check so we reduce the amount of vars to track.